internal/mode/static/state/graph/grpcroute.go

```diff
@@ -0,0 +1,204 @@
+package graph
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/conditions"
+	staticConds "github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/validation"
+)
+
+// GRPCRoute represents a GRPCRoute.
+type GRPCRoute struct {
```

Comment:
I wonder if we can use a common type to represent both HTTP and GRPCRoutes.

I think this way we remove a lot of duplicated code. For example, `addHTTPBackendRefsToRules` and `addGRPCBackendRefsToRules` in `backend_refs.go` and `upsertHTTPRoute` and `upsertGRPCRoute` in `configuration.go`.

I was thinking about something like below:
```
type L7Route struct {
	Source     client.Object
	Spec       L7RouteSpec
	ParentRefs []ParentRef
	Conditions []conditions.Condition
	Valid      bool
	Attachable bool
}

type L7RouteSpec struct {
	Hostnames []v1.Hostname
	Rules     []Rule
}

type Rule struct {
	Matches      []v1.HTTPRouteMatch
	ValidMatches bool
	ValidFilters bool
	Filters      []any
	BackendRefs  []BackendRef
}

type BackendRef struct {
	v1.BackendRef
	Filters []any
}
```

Note that we will need to convert GPRC match into HTTPMatch, but this seems OK, since GRPC match is a subset and syntactic sugar of HTTPMatch.